### DISCUSSION TITLE
Possible to support both CDC-ECM and RNDIS ?

### DISCUSSION THREAD

**Discussion by Mowglli** [hathach/tinyusb#1998] [Q&A]
Hello, i have configured the ESP32-S3 to pop up as RNDIS device with IP and everything, i can do the same with CDC-ECM, if i change that in the sdk config. 
Is it possible to support both, without having to compile with the other setting ? 
Can the tud_descriptor_configuration_cb(), be used?

**Comment by hathach** [comment]
yes, they are literally the same driver, check out the exammple that use both rndis and ecm https://github.com/hathach/tinyusb/blob/master/examples/device/net_lwip_webserver/src/usb_descriptors.c

**Reply by Mowglli** [reply]
@hathach 

Okay thank you ill have a look at that.

What about iPad and Android phones? Is there somehow support for this? I have tried to day that it accepts 500mA, the phone et Can then Power the device put does not get an ip☺️

**Comment by Mowglli** [comment]
@hathach 
Got it to work now, thanks!
